PyInstaller/utils/hooks/__init__.py

```diff
@@ -984,3 +986,61 @@ def relpath_to_config_or_make(filename):
     # Relative path in the dist directory.
     prefix = _find_prefix(filename)
     return os.path.relpath(os.path.dirname(filename), prefix)
+
+
+def get_typelibs(module, version):
+    datas = []
+    statement = """
+import gi
+gi.require_version("GIRepository", "2.0")
+from gi.repository import GIRepository
+print(GIRepository.Repository.get_search_path())"""
+    typelibs_path = eval_statement(statement)[0]
+    pattern = os.path.join(typelibs_path, module + '*' + version + '*')
+    for f in glob.glob(pattern):
+        datas.append(gir_library_path_fix(f))
+    return datas
+
+
+def gir_library_path_fix(path):
+    import subprocess
+
+    path = os.path.abspath(path)
+    common_path = os.path.commonprefix([sys.prefix, path])
+    gir_path = os.path.join(common_path, 'share', 'gir-1.0')
+
+    typelib_name = os.path.basename(path)
+    gir_name = os.path.splitext(typelib_name)[0] + '.gir'
+
+    gir_file = os.path.join(gir_path, gir_name)
+
+    if is_darwin:
+        if os.path.exists(gir_path):
+            if os.path.exists(gir_file):
+                with open(gir_file, 'r') as f:
+                    lines = f.readlines()
+
+                with open(os.path.join(tempfile.gettempdir(), gir_name), 'w') as f:
+                    for line in lines:
+                        if 'shared-library' in line:
+                            split = re.split('(=)', line)
+                            files = re.split('(["|,])', split[2])
+                            for count, item in enumerate(files):
+                                if 'lib' in item:
+                                    files[count] = '@loader_path/' + os.path.basename(item)
+                            line = ''.join(split[0:2]) + ''.join(files)
+                        f.write(line)
+
+                command = subprocess.Popen(('g-ir-compiler', os.path.join(tempfile.gettempdir(), gir_name),
+                                            '-o', os.path.join(tempfile.gettempdir(), typelib_name)))
```

Comment:
What a dull tool :-( Pleas put this in a comment, too, others may ask this, too.Thanks.